src/clusterfuzz/_internal/bot/tasks/update_task.py

```diff
@@ -251,15 +251,15 @@ def update_source_code():
 
     try:
       # Make sure we can override files without prior write permission
+      target_destination = os.path.join(cf_source_root_parent_dir, file.name)
+      try:
+        os.chmod(target_destination, 0o755)
+      except FileNotFoundError:
```

Comment:
Why would this happen?

Reply:
Suppose we have proto 3.20.3, which does not have, lets say, google/proto/wrapper.py.
If we get 4.23.4 in the update task, we would be calling:
```
vitorguidi@vitorguidi1:~/Desktop$ python3
Python 3.11.8 (main, May 26 2024, 00:40:16) [GCC 13.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import os
>>> os.chmod('google/proto/wrapper/py', 0o755)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
FileNotFoundError: [Errno 2] No such file or directory: 'google/proto/wrapper/py'
>>> 
```

This handles the case were there will not be a file to be overwritten.

Reply:
We probably should be nuking directories so we don't accidentally leave stale files in right?

Reply:
You can merge this for now, but I think we really should do a better fix here.

Reply:
IMO if we are ever tackling this, we should be making a push for immutable infrastructure. Nuking directories to fix the update task is tackling the symptom, not the cause of the issue

Reply:
Also, we know replacing files has worked up to this point. We do not know if nuking files/directories will introduce bugs we have not faced so far, I would rather keep this approach than take the risk and spend extra time debugging new failure modes.

Reply:
Merge for now.